lib/openshift/cluster_version.rb

```diff
@@ -1,19 +1,70 @@
 module BushSlicer
   class ClusterVersion < ClusterResource
     RESOURCE = "clusterversions"
-    def channel(user: nil, cached: true, quiet: false)
+    def channel(user: nil, cached: false, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('spec', 'channel')
     end
 
-    def upstream(user: nil, cached: true, quiet: false)
+    def upstream(user: nil, cached: false, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('spec', 'upstream')
     end
 
-    def version(user: nil, cached: true, quiet: false)
+    def version(user: nil, cached: false, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('status', 'desired', 'version')
     end
+
+    def history(user: nil, cached: false, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      rr.dig('status', 'history')
+    end
+
+    # return a list of history matching the state we want to filter on
+    # query_hash is the parameter we want to filter on,
+    def history_matching(query_hash: nil , user: nil, cached: true, quiet: false)
+      key = query_hash.keys.first
+      value = query_hash[key]
```

Comment:
Can we instead have `key` and `value` parameters to the method instead of this `query_hash` parameter that makes less clear what should it contain?

Reply:
done